primitives/state-machine/src/trie_backend_essence.rs

```diff
@@ -226,12 +222,23 @@ impl<S: TrieBackendStorage<H>, H: Hasher, C> TrieBackendEssence<S, H, C> {
 			empty: H::hash(&[0u8]),
 			#[cfg(feature = "std")]
 			cache: Arc::new(RwLock::new(Cache::new())),
-			#[cfg(feature = "std")]
 			trie_node_cache: None,
 			#[cfg(feature = "std")]
 			recorder: None,
-			#[cfg(not(feature = "std"))]
-			_phantom: PhantomData,
+		}
+	}
+
+	/// Create new trie-based backend.
+	pub fn new_with_cache(storage: S, root: H::Out, cache: Option<C>) -> Self {
```

Comment:
You could simplify `TrieBackendEssence::new` by calling this method in there. (`Self::new_with_cache(storage, root, None)` instead of having the whole struct initialized in two places)

Reply:
Also third copy below could be initialized with new.